File: 0657-robot-return-to-origin/0657-robot-return-to-origin.py
class Solution:
    def judgeCircle(self, moves: str) -> bool:
      
      dirVal = {
        "U":1,
        "R":200,
        "L":-200,
        "D":-1
      }
      
      ans = 0
      for i in moves:
        ans += dirVal.get(i)
        print(ans, end=' ')
     
      
      return ans == 0
        
        
        

File: 0657-robot-return-to-origin/0657-robot-return-to-origin.c
bool judgeCircle(char * moves){
  
  int u = 0;
  int d = 0;
  int r = 0;
  int l = 0;
  
  for(int i=0; i<strlen(moves); i++){
    
    if (moves[i] == 'U'){
      u += 1;
    }
    
    else if (moves[i] == 'D'){
      d += 1;
    }    
    
    else if (moves[i] == 'L'){
      l += 1;
    }    
    
    else if (moves[i] == 'R'){
      r += 1;
    }
  }
  
  return l == r && u == d;

}